### DISCUSSION TITLE
Tasks Failing immediately after queued and Showing Log file does not exist

### DISCUSSION THREAD

**Discussion by devashishpatil56** [apache/airflow#20808] [Q&A]
Hi,

I am facing this problem where tasks are failing intermittently right after they are queued and when I check logs I see something like this:

`*** Log file does not exist: /opt/airflow/airflow/logs/<DAG_NAME>/<TASK_NAME>/2022-01-10T04:00:00+00:00/12.log
*** Fetching from: http://:8793/log/<DAG_NAME>/<TASK_NAME>/2022-01-10T04:00:00+00:00/12.log
*** Failed to fetch log file from worker. Request URL is missing an 'http://' or 'https://' protocol.`

When I check the logs for the scheduler I can see this:

`ERROR/ForkPoolWorker-15] Failed to execute task dag_id could not be found: <DAG_ID>. Either the dag did not exist or it failed to parse..`

When I check the Task Instance Details, the hostname is blank.

Setup:
Airflow Version: 2.1.0
Database: postgres9.6
Using CeleryExecuter and Redis and hosted on Kubernetes. Also, using pgbouncer for connection pooling

Additional:
We have multiple Sensor tasks running in reschedule mode. This issue is more frequent with the sensor tasks but can be seen intermittently with other types of tasks as well.
We have put multiple retries as well, but very often this happens for all the subsequent retries as well.

**Comment by jedcunningham** [comment]
How are you deploying DAGs? Is it possible the scheduler and worker(s) have different sets of DAGs?

**Reply by devashishpatil56** [reply]
I am bundling the dags in the docker image itself and both scheduler and worker are running on that same image.

**Reply by potiuk** [reply]
It almost look like you have another scheduler running an image that has no DAGs embedded. In any way without more logs I am afraid we won't be able to help.

**Reply by devashishpatil56** [reply]
I am putting the Dags along with the docker image and all the pods share the same image. 

I had been seeing DAG parsing errors with this one, I increased the DAG parsing timeouts along with a few other variables and this seems to help here. 

```
- name: AIRFLOW__CORE__MIN_SERIALIZED_DAG_UPDATE_INTERVAL
  value: "180"
- name: AIRFLOW__CORE__MIN_SERIALIZED_DAG_FETCH_INTERVAL
  value: "90"
- name: AIRFLOW__CORE__DAGBAG_IMPORT_TIMEOUT
  value: "90"
- name: AIRFLOW__CORE__DAG_FILE_PROCESSOR_TIMEOUT
  value: "180"
```

Along with this one, I also reduced the worker concurrency to 8 from the default 16. 
All this configuration change helped a lot in performance over the past 1 week and also helped with this issue as well: https://github.com/apache/airflow/discussions/20810

**Reply by potiuk** [reply]
I believe the problem you had could be connected with lack of memory you had in K8s or other resources (but memory is most likely). I woudl really advice you to look at the history of your K8S/Kubelet logs to see if this happened

**Reply by potiuk** [reply]
I'd say - you had memory problems (but this is wild guess).

**Reply by devashishpatil56** [reply]
I have checked the memory for the pods, over-provisioned it for the last couple of weeks, the average usage is pretty low and the max usage is also reaching around 30%. Same for the pod CPU as well. But I can still see the issues intermittently.

Adding a few more logs here:
```
[2022-02-09 06:36:18,091: ERROR/ForkPoolWorker-7] Task airflow.executors.celery_executor.execute_command[018130f9-873f-42f9-b9d3-17f5e006f02d] raised unexpected: AirflowException('Celery command failed on host: worker')
Traceback (most recent call last):
File "/home/airflow/.local/lib/python3.8/site-packages/celery/app/trace.py", line 412, in trace_task
R = retval = fun(*args, **kwargs)
File "/home/airflow/.local/lib/python3.8/site-packages/celery/app/trace.py", line 704, in __protected_call__
return self.run(*args, **kwargs)
File "/home/airflow/.local/lib/python3.8/site-packages/airflow/executors/celery_executor.py", line 87, in execute_command
_execute_in_fork(command_to_exec)
File "/home/airflow/.local/lib/python3.8/site-packages/airflow/executors/celery_executor.py", line 98, in _execute_in_fork
raise AirflowException('Celery command failed on host: ' + get_hostname())
airflow.exceptions.AirflowException: Celery command failed on host: worker
```

`{scheduler_job.py:1241} ERROR - Executor reports task instance <TaskInstance: DAG_ID.TASK_ID 2022-01-10 04:00:00+00:00 [queued]> finished (failed) although the task says its queued. (Info: None) Was the task killed externally?
`

**Reply by potiuk** [reply]
There were a number of related problems fixed in 2.2.1/2/3 - i suggest to upgrade and check if the problem has already been fixed there.

**Reply by potiuk** [reply]
You might wait short time for 2.2.4 thought (it is coming in a few weeks)

**Reply by changxiaoju** [reply]
> How are you deploying DAGs? Is it possible the scheduler and worker(s) have different sets of DAGs?

i think i am facing this problem. it was all ok at beginning, and then,i copied a new dag with the same name to replace the old dag in airflow/dags/, and then i face the same error as jedcunningham. however, i do not know how to fix it.
i restarted webserver and scheduler and nothing worked, there are still no log

**Reply by Leoduv** [reply]
> I am putting the Dags along with the docker image and all the pods share the same image.
> 
> I had been seeing DAG parsing errors with this one, I increased the DAG parsing timeouts along with a few other variables and this seems to help here.
> 
> ```
> - name: AIRFLOW__CORE__MIN_SERIALIZED_DAG_UPDATE_INTERVAL
>   value: "180"
> - name: AIRFLOW__CORE__MIN_SERIALIZED_DAG_FETCH_INTERVAL
>   value: "90"
> - name: AIRFLOW__CORE__DAGBAG_IMPORT_TIMEOUT
>   value: "90"
> - name: AIRFLOW__CORE__DAG_FILE_PROCESSOR_TIMEOUT
>   value: "180"
> ```
> 
> Along with this one, I also reduced the worker concurrency to 8 from the default 16. All this configuration change helped a lot in performance over the past 1 week and also helped with this issue as well: #20810

Thank you, fixed it for me! I was stuck on this `Either the dag did not exist or it failed to parse` on my k8s deployed Airflow, I tried everything. A "Maybe check parsing timeouts" would have been very helpful in the error message 🤷
I'll try giving more CPU to my pods and see if it improves the DAG parsing.

Cheers

**Reply by MattiaGallegati** [reply]
I have just installed Airflow on Kubernetes using the official Helm, version 2.3.2, same configuration: external Redis + external PG + PG Bouncer and use CeleryExecutor.
The installation is completely unstable due to some errors that I'm trying to solve and report.
One of the errors is  the one that you have pointed out.

**Reply by potiuk** [reply]
@MattiaGallegati - could you please report details of YOUR problems and YOUR logs in separate issues? This is the only way you can  get help (And most of the case the problem is with your deployment, so YOUR logs and investigation is needed to diagnose it. Often those are with memory, or other resources missing - for example if you try to run a lot of stuff on a small machine, things will not work for sure. Please report your issue once you investigate it enough to be able to show a reproducible case, or if not - start a new discussion, but provide your logs. Sometimes even if logs look similar, it means diffferent error - and often those errors depend on your particular configuration and circumstances.

So I encourage you to report the issues with enough details so that people here can help (mind that they are helping for free in their free time, so the more details you provide to help them to help you - the better.

**Reply by MattiaGallegati** [reply]
Yes of course. As I said I'm trying to gather evidence and do some research on already opened issue in order to see if my problem are already known or solved. Everything is running in a distributed-production like environment.
I will report everything I cannot solve for sure, once I've made some tests on it, meanwhile I'm trying also to help others in the free time.

**Comment by changxiaoju** [comment]
> devashishpatil56

sorry, the same error with devashishpatil56

**Comment by IvanAlekhin** [comment]
Faced with this error `Failed to execute task dag_id could not be found` too. 
Check, if your DAG code in $AIRFLOW_HOME/dags in the celery worker instance. Helped for me.
